samples/metrics_monitor/src/cttmetrics_i915_pmu.cpp

```diff
@@ -518,13 +529,35 @@ cttStatus CTTMetrics_PMU_GetValue(unsigned int count, float* out_metric_values)
             group = &g_ctx.pm.groups[pm_metric_idx];
 
             value = perf_elapsed(&group->metrics[I915_SAMPLE_BUSY]);
+            time = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_BUSY]);
+
+            double value_sema, value_wait;
+            double time_sema, time_wait;
+            bool is_sema_wait_available = is_engine_config(group->metrics[I915_SAMPLE_BUSY].config) && (3 == group->num_metrics);
+
+            if (is_sema_wait_available) {
+                value_sema = perf_elapsed(&group->metrics[I915_SAMPLE_SEMA]);
+                time_sema = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_SEMA]);
+
+                value_wait = perf_elapsed(&group->metrics[I915_SAMPLE_WAIT]);
+                time_wait = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_WAIT]);
+            }
+
             if (g_ctx.metrics[metric_idx] == CTT_AVG_GT_FREQ) {
                 value *= 1000000000; // to compensate time in ns
             }
             else {
                 value *= 100; // we need %
+                value_sema *= 100;
+                value_wait *= 100;
+            }
+
+            value /= time;
+            if (is_sema_wait_available) {
+                value_sema /= time_sema;
+                value_wait /= time_wait;
+                value -= value_sema + value_wait; // Substract sema and wait from busy metric
```

Comment:
We should do that only if i915 supports preemption. Otherwise, we need to count sema and wait as a time engine is really occupied.